### PR TITLE
Return reponse header in error condition

### PR DESCRIPTION
**Pull request by gadenbuie**
This small PR returns the response headers as `$response` in the error condition thrown from `gh_error()` to make it possible to use information returned in the headers by GitHub when handling the error.

In particular, this is useful when reaching rate limits as it would allow users to use the `Retry-After` or `X-RateLimit-Reset` properties to wait for the rate limit to reset or the retry-after period to pass.

```r
# this search will quickly make too many requests per minute
# and trigger the abuse rate limit
err <- tryCatch(
  gh("/search/code", q = "foo", .limit = Inf),
  "http_error_403" = identity
)
err
#> <github_error in gh_process_response(raw): 
#> GitHub API error (403): 403 Forbidden
#> Message: You have triggered an abuse detection mechanism. Please wait a few minutes before you try again.
#> Read more at https://developer.github.com/v3/#abuse-rate-limits>

str(err$response[c("status", "retry-after")])
#> List of 2
#>  $ status     : chr "403 Forbidden"
#>  $ retry-after: chr "60"
```

I used `response` to align with the attribute name in a successful response.

```r
x <- gh("/repos/r-lib/gh")
attributes(x)$response
#> ... response headers
```

### CODE REVIEW

**Comment by gadenbuie** [comment]
I renamed the headers item `response_headers` and added the response content as `response_content`.

I hope it's okay, but I rolled #116 into this PR as it also touches `gh_error()` and resolves an issue that occurs with conditional requests that don't return a response body. (In short, when the body is empty `res` is `raw(0)` so `gh_error()` fails when trying to access `res$message`.)

File: R/gh_response.R [hunk 1]
```diff
@@ -44,6 +44,10 @@ gh_error <- function(response, call = sys.call(-1)) {
   res <- content(response)
   status <- status_code(response)
 
+  if (!length(res)) {
+    res <- list(message = "Empty Body\n")
+  }
+
```
Comment (jennybc) [inline R/gh_response.R:50]: Small drive-by observation:I may be taking this out of context.

Should this really be manufacturing a body / a string, where one did not previously exist?

In terms of non-error response, we've already had one round of rethink on this in gh, where we went from returning the empty string to a list of length zero. And that ended up being a breaking change for usethis (but a positive change) Just want to make sure we don't make the same mistake again.

https://github.com/r-lib/gh/commit/6b735b192f44b7f7253a3a6cf7827ccab1060d54

Comment (gadenbuie) [reply R/gh_response.R:50]: Another reason why I pulled this change into this PR is to make sure that the response content is returned _as is_ in the condition. `gh_error()` expects the response to be a list with a `message` item which it uses to construct the output message. After this PR, you won't need to look at the message to reason about the response, instead both the headers and the content are returned unprocessed in the condition object -- this is the intent behind calling `content(response)` again in the condition construction at the end of `gh_error()`.

I was trying to make minimal changes, but I'd be happy to refactor if you prefer another approach.

Comment (jennybc) [reply R/gh_response.R:50]: I did not take in the whole picture, so definitely take my comment with a large grain of salt.

**Comment by gaborcsardi** [comment]
Thanks again and sorry for the long wait! I removed #116 because I think we handle that elsewhere now.

**Pull request merged**